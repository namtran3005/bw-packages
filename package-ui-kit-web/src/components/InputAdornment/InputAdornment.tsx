import * as React from 'react';
import { InputAdornment as MuiInputAdornment } from '@material-ui/core';
import { InputAdornmentProps as MuiInputAdornmentProps } from '@material-ui/core/InputAdornment';
interface InputAdornmentProps extends MuiInputAdornmentProps {
  innerRef?: any; // eslint-disable-line @typescript-eslint/no-explicit-any
}

const InputAdornment: React.FunctionComponent<InputAdornmentProps> = (
  props
) => <MuiInputAdornment {...props} />;

export { InputAdornment, InputAdornmentProps };
