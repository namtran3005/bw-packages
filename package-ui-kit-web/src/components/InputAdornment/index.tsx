export { InputAdornment, InputAdornmentProps } from './InputAdornment';
