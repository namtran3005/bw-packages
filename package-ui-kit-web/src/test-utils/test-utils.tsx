import {
  render as rtlRender,
  waitFor,
  RenderOptions as RTLRenderOptions,
} from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import userEvent from '@testing-library/user-event';
import { WebAppThemeProvider } from '../theme';
import '@testing-library/jest-dom';
import '@testing-library/jest-dom/extend-expect';

type RenderOptions = RTLRenderOptions;

function renderWithTestHarness(
  ui: React.ReactElement,
  renderOptions: RenderOptions = {}
) {
  function Wrapper({ children }: any) {
    return (
      <MemoryRouter>
        <WebAppThemeProvider>{children}</WebAppThemeProvider>
      </MemoryRouter>
    );
  }

  return rtlRender(ui, {
    wrapper: Wrapper,
    ...renderOptions,
  });
}

// Use if you're in a hurry (better than nothing)
// be sure to verify no memory leak from implicit
// waitFor by adding these flags to jest: --runInBand --detectOpenHandles
async function renderAct(ui: React.ReactElement, options: RenderOptions = {}) {
  const view = renderWithTestHarness(ui, options);
  return await waitFor(() => view, {
    timeout: 5000,
    onTimeout: (e) => {
      // eslint-disable-next-line no-console
      console.error(e);
      return e;
    },
  });
}

export * from '@testing-library/react';

export { renderWithTestHarness, renderAct, userEvent };
