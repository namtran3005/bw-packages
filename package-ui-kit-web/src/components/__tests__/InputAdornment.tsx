import * as React from 'react';

import { renderWithTestHarness } from '../../test-utils/test-utils';
import { InputAdornment } from '../InputAdornment';

const defaultProps = {
  // eslint-disable-next-line @typescript-eslint/prefer-as-const
  position: 'start' as 'start',
};

describe('InputAdornment', () => {
  it('should render input label', () => {
    const { baseElement } = renderWithTestHarness(
      <InputAdornment {...defaultProps} />
    );
    expect(baseElement).toMatchSnapshot();
  });
});
