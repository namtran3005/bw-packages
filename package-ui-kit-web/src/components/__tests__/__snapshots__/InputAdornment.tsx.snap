// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputAdornment should render input label 1`] = `
<body>
  <div>
    <div
      class="MuiInputAdornment-root MuiInputAdornment-positionStart"
    />
  </div>
</body>
`;
